gitpod-io/gitpod
# validate jb warmup

## Description


This PR extract JB warmup task configuration from server to JB integration itself to enable usage of `gp validate` to investigate warmup issues in the regular workspace. Composition of IDE prebuild tasks are done by supervisor, i.e. it can detect multiple desktop IDEs now. 

## Related Issue(s)

A prerequisite to IDE-182

## How to test


- Create a project for https://github.com/akosyakov/spring-petclinic
- And run a prebuild
- It does not need to succeed, but just check that warmup of IntelliJ was triggered from logs
- Or test that starting IntelliJ from https://ak-validat3f63c69ca3.preview.gitpod-dev.com/projects/spring-petclinic/766afbb5-ec90-46e9-906b-d56fa6b7d9c9 works both for stable and latest.

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ak-validat3f63c69ca3</li>
	<li><b>🔗 URL</b> - <a href="https://ak-validat3f63c69ca3.preview.gitpod-dev.com/workspaces" target="_blank">ak-validat3f63c69ca3.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold
